ci/get_hdf5_if_needed.sh

```diff
@@ -54,9 +61,15 @@ else
         fi
 
         pushd /tmp
-        #                                   Remove trailing .*, to get e.g. '1.12' ↓
-        curl -fsSLO "https://www.hdfgroup.org/ftp/HDF5/releases/hdf5-$MAJOR_V.$MINOR_V/hdf5-$HDF5_VERSION/src/hdf5-$HDF5_VERSION.tar.gz"
-        tar -xzvf hdf5-$HDF5_VERSION.tar.gz
+        # temporarily turn off -e option as curl will emit non-zero exit code
+        # as only one assert format is valid
+        # (curl exits with 22 with http code >400)
+        set +e
+        curl -fsSL -o "hdf5-$HDF5_VERSION.tar.gz" "https://github.com/HDFGroup/hdf5/archive/refs/tags/{${ASSET_FMT1},${ASSET_FMT2}}.tar.gz"
+        set -e
+
+        tar -xzvf hdf5-$HDF5_VERSION.tar.gz --one-top-level --strip-components=1
```

Comment:
It looks like the Mac version of the `tar` command doesn't have the `--one-top-level` flag, unfortunately.

Reply:
@takluyver Thanks for pointing this out, amended.